plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetFileWriter.java

```diff
@@ -89,6 +89,11 @@ public long getWrittenBytes()
         return parquetWriter.getWrittenBytes();
     }
 
+    public long getBufferedBytes()
```

Comment:
Why we want to use it for Iceberg, but not for Hive?

Reply:
As far as I saw hive does it little differently but it also takes bufferedbytes into account, just not directly

Reply:
i am looking at
https://github.com/trinodb/trino/blob/9ab09e99d6c56fe955697a51a95400c367525133/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSink.java#L369

and indeed ORC includes buffered bytes, and Parquet does not (https://github.com/trinodb/trino/pull/10957#discussion_r801497102)

Still, we don't need this to be different for Hive vs Iceberg, right?

Reply:
Ok sorry for a wrong explanation, so Hive uses implementations of `FileWriter`. There are many of them and as far as I understand it is hard for some of them to measure even already written bytes. For example here https://github.com/trinodb/trino/blob/6588a9e4f5e8489330cb013476988b547efc8d8d/plugin/trino-hive/src/main/java/io/trino/plugin/hive/RecordFileWriter.java#L135-L157

I had no idea how to implement `getWrittenBytes` and `getBufferedByte` for some `FileWriter` implementations so I only focused on those used by Iceberg.